test/scripts/traffic-flow-tests.sh

```diff
@@ -0,0 +1,167 @@
+            persistent: "false"
+            sriov: "false"
+        client:
+          - name: "ovn-worker2"
+            sriov: "false"
+        plugins: []
+kubeconfig:
+kubeconfig_infra:
+EOT
+}
+
+process_test_results() {
+    local result_file="$1"
+    local unique_tests_script='.["tft-tests"][]["flow_test"].tft_metadata.test_case_id'
+    local failures_script='
+      .["tft-tests"][] |
+      if .["flow_test"].success then empty
+      else {test_id: .["flow_test"].tft_metadata.test_case_id, message: .["flow_test"].msg}
+      end
+    '
+
+    log "INFO" "Processing test results from: ${result_file}"
+
+    local unique_test_ids
+    unique_test_ids=$(jq -r "${unique_tests_script}" "${result_file}" | sort -u)
+
+    echo "Test Cases:"
+    echo "-----------------------------------------------------"
+    echo "${unique_test_ids}"
+
+    local failures
+    failures=$(jq -c "${failures_script}" "${result_file}")
+
+    if [ -z "${failures}" ]; then
+        echo "-----------------------------------------------------"
+        echo "All traffic flow tests succeeded."
+    else
+        echo "-----------------------------------------------------"
+        echo "Some traffic flow tests failed. Details:"
+        echo "${failures}" | jq .
+        exit 1
+    fi
+}
+
+setup() {
+    check_dependencies
+
+    "${OCI_BIN}" pull "${TFT_TEST_IMAGE}" || error_exit "Failed to pull the test image."
+    local KIND_CLUSTER_NAME
+    KIND_CLUSTER_NAME=$(kind get clusters)
+    kind load docker-image "${TFT_TEST_IMAGE}" --name "${KIND_CLUSTER_NAME}" || error_exit "Failed to load the test image."
+
+    if [ -d "${TRAFFIC_FLOW_TESTS_FULL_PATH}" ]; then
+        error_exit "Install folder already exists: ${TRAFFIC_FLOW_TESTS_FULL_PATH}"
+    fi
+    mkdir -pv "${TRAFFIC_FLOW_TESTS_FULL_PATH}"
+    cd "${TRAFFIC_FLOW_TESTS_FULL_PATH}"
+
+    git init
+    git remote add origin "${TRAFFIC_FLOW_TESTS_REPO}"
+    git fetch --depth=1 origin "${TRAFFIC_FLOW_TESTS_COMMIT}"
+    git checkout "${TRAFFIC_FLOW_TESTS_COMMIT}"
+
+    python -m venv tft-venv
+    source tft-venv/bin/activate
+    pip install --upgrade pip
+    pip install -r requirements.txt
+
+    local CONFIG_FILE="config.yaml"
+    create_config_file_defaults "$CONFIG_FILE"
+
+    sed -i "s|^kubeconfig:.*|kubeconfig: \"$KUBECONFIG\"|" "$CONFIG_FILE"
+    sed -i "s|test_cases:.*|test_cases: \"$TRAFFIC_FLOW_TESTS\"|" "$CONFIG_FILE"
+
+    log "INFO" "Setup complete. Configuration file created: $CONFIG_FILE"
+
+    echo "---"
+    cat "$CONFIG_FILE"
+}
+
+run() {
+    cd "${TRAFFIC_FLOW_TESTS_FULL_PATH}" || error_exit "Run folder not found. Missing setup?"
+    source tft-venv/bin/activate || error_exit "Python environment missing. Missing setup?"
+
+    time python main.py config.yaml || error_exit "Test execution FAILED."
+
+    local RESULT_FILE
+    RESULT_FILE=$(ls -rt "${TRAFFIC_FLOW_TESTS_FULL_PATH}/ft-logs/"*.json | tail -1) || error_exit "Unable to locate results.json."
```

Comment:
This looks cumbersome. It should be easier to find the output file. With https://github.com/wizhaoredhat/ocp-traffic-flow-tests/pull/164 , you could write
```
    local OUTPUT_BASE="$TRAFFIC_FLOW_TESTS_FULL_PATH/ft-logs/result-"
    time python main.py config.yaml --o "$OUTPUT_BASE" || error_exit "Test execution FAILED."
    local RESULT_FILE="${OUTPUT_BASE}000.json"
```

Reply:
I still don't want to assume that the file is called `"${OUTPUT_BASE}000.json"` so I incorporated your changes but am wondering if this is still too cumbersome. wdyt?

Reply:
> I still don't want to assume that the file is called "${OUTPUT_BASE}000.json" so I incorporated your changes but am wondering if this is still too cumbersome. wdyt?

`ls -rt "${OUTPUT_BASE}"*.json | tail -1` will definitely work in your case (since there won't be any other such files).

I think it would be also safe to anticipate the exact file name to be `*000.json`. That's literally the point of the option. But being more flexible is fine.